packages/@lwc/engine-core/src/framework/upgradable-element.ts

```diff
@@ -42,6 +60,19 @@ export function getUpgradableConstructor(
             }
         }
     };
+    if (features.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+        CE.prototype.connectedCallback = function () {
+            if (checkHasVM(this)) {
```

Comment:
Can this check be moved to L60, just after the `if` block?
The constructor has to be called always before connectedCallback and disconnectedCallback. Moving the check these will reduce it to running in one time.

Reply:
Actually I checked, and it looks like we can't move this to the constructor, because the `this` is not in the `WeakSet` yet. We can move the check to the first `connectedCallback` though.

Reply:
OK, we can move it to the constructor, but now the tests are emitting a lot more warnings. That's fine though.